framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1387,148 +1425,173 @@ private AnnotationMirror getAnnotation(
-                                + " and expected: "
-                                + expected);
+            return convert(((IntegerLiteralExpr) expr).asInt(), valueKind);
+        } else if (expr instanceof LongLiteralExpr) {
+            // No convert if the expression is a long, the annotation value must be a long, too.
+            return ((LongLiteralExpr) expr).asLong();
+        } else if (expr instanceof ClassExpr) {
+            ClassExpr classExpr = (ClassExpr) expr;
+            String className = classExpr.getType().toString();
+            if (importedTypes.containsKey(className)) {
+                return importedTypes.get(className).asType();
             }
-        } else if (expr instanceof StringLiteralExpr) {
-            StringLiteralExpr slexpr = (StringLiteralExpr) expr;
-            ExecutableElement var = builder.findElement(name);
-            TypeMirror expected = var.getReturnType();
-            if (expected.getKind() == TypeKind.DECLARED) {
-                builder.setValue(name, slexpr.getValue());
-            } else if (expected.getKind() == TypeKind.ARRAY) {
-                String[] arr = {slexpr.getValue()};
-                builder.setValue(name, arr);
-            } else {
-                ErrorReporter.errorAbort(
-                        "StubParser: unhandled annotation attribute type: "
-                                + slexpr
-                                + " and expected: "
-                                + expected);
+            Class<?> clazz;
+            try {
+                clazz = Class.forName(className);
+            } catch (ClassNotFoundException e) {
+                // TODO: should this be stubAlwaysWarn, as it is an input stub file error?
+                ErrorReporter.errorAbort("StubParser: unknown class name " + className);
+                throw new Error("dead code; this can't happen");
             }
-        } else if (expr instanceof ArrayInitializerExpr) {
-            ExecutableElement var = builder.findElement(name);
-            TypeMirror expected = var.getReturnType();
+            // TODO: is it worth putting this value back into importedTypes?
+            return TypesUtils.typeFromClass(
+                    atypeFactory.getContext().getTypeUtils(),
+                    atypeFactory.getElementUtils(),
+                    clazz);
+
+        } else if (expr instanceof NullLiteralExpr) {
+            // TODO: should this be stubAlwaysWarn, as it is an input stub file error?
+            ErrorReporter.errorAbort(
+                    "Null found as value for %s. Null isn't allowed as an annotation value", name);
+            return null; // dead code
+        } else {
+            ErrorReporter.errorAbort("Unexpected annotation expression: " + expr);
+            return null; // dead code
+        }
+    }
+
+    /**
+     * Converts {@code number} to {@code expectedKind}.
+     * <p>
+     * {@code @interface Anno { long value();})
```

Comment:
I changed the annotation attribute type to `long`. Otherwise the comment below on line 1500 doesn't make sense to me. If I'm misunderstanding, can you clarify what is happening?

Reply:
Yes, `int` should have been `long`. Thanks!